### PR TITLE
Validate required namespaces.

### PR DESCRIPTION
**Pull request by jortel**
Validate required namespaces starting with the `velero` namespace.

### CODE REVIEW

File: pkg/controller/migplan/validation.go [hunk 1]
```diff
@@ -43,18 +46,20 @@ const (
...
+	InvalidStorageRefMessage              = "The `migStorageRef` must reference a `migstorage`."
+	InvalidAssetCollectionRefMessage      = "The `migAssetCollectionRef` must reference a `migassetcollection`."
+	SourceClusterNotReadyMessage          = "The referenced `srcMigClusterRef` does not have a `Ready` condition."
+	DestinationClusterNotReadyMessage     = "The referenced `dstMigClusterRef` does not have a `Ready` condition."
+	StorageNotReadyMessage                = "The referenced `migStorageRef` does not have a `Ready` condition."
+	AssetCollectionNotReadyMessage        = "The referenced `migAssetCollectionRef` does not have a `Ready` condition."
+	AssetNamespaceNotFoundMessage         = "The following asset `namespaces` [%s] not found on the source cluster."
+	InvalidDestinationClusterMessage      = "The `srcMigClusterRef` and `dstMigClusterRef` cannot be the same."
+	EnsureStorageFailedMessage            = "Failed to create/validate backup and volume snapshot storage."
+	NsNotFoundOnSourceClusterMessage      = "The follow namespaces [%s] not found on source cluster."
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: Assume this was meant to read:

`The following namespaces`

I think we could reword this to 

`Namespaces [%s] not found on source cluster`

File: pkg/controller/migplan/validation.go [hunk 2]
```diff
@@ -43,18 +46,20 @@ const (
...
+	NsNotFoundOnDestinationClusterMessage = "The follow namespaces [%s] not found on destination cluster."
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: Same thing here

File: pkg/controller/migplan/validation.go [hunk 3]
```diff
@@ -329,3 +341,111 @@ func (r ReconcileMigPlan) validateDestinationCluster(plan *migapi.MigPlan) (int,
...
+		})
+		return 1, nil
+	}
+
+	return 0, nil
+}
+
+// Validate required namespaces on the destination cluster.
+// Returns error and the total error conditions set.
+func (r ReconcileMigPlan) validateDestinationNamespaces(plan *migapi.MigPlan) (int, error) {
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: What are the "required namespaces" here? Velero will create any namespaces we are migrating as part of the "Restore" process.

Comment (jortel) [reply pkg/controller/migplan/validation.go]: For both the source and destination, the required namespaces are not related to asset collection.  Instead, the list (to start with) includes the namespace the controller will create velero resources in.  Since it's intended to be configurable, It may not be `velero`.

File: pkg/controller/migplan/validation.go [hunk 4]
```diff
@@ -329,3 +341,111 @@ func (r ReconcileMigPlan) validateDestinationCluster(plan *migapi.MigPlan) (int,
...
+	}
+	totalSet += nSet
+
+	return totalSet, nil
+}
+
+// Validate required namespaces on the source cluster.
+// Returns error and the total error conditions set.
+func (r ReconcileMigPlan) validateSourceNamespaces(plan *migapi.MigPlan) (int, error) {
+	namespaces := []string{velerorunner.VeleroNamespace}
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: Ah, I see now

File: pkg/controller/migplan/validation.go (hunk 3 above)
Comment (djwhatle) [reply pkg/controller/migplan/validation.go]: I see that now, thanks for the explanation.

File: pkg/controller/migplan/validation.go (hunk 1 above)
Comment (jortel) [reply pkg/controller/migplan/validation.go]: Fixed.
